src/main/scala/org/bitcoins/core/protocol/CompactSizeUInt.scala

```diff
@@ -22,22 +28,37 @@ trait CompactSizeUInt {
 
   def hex = size match {
     case 1 => if (num.toHexString.size == 1) "0" + num.toHexString else num.toHexString
-    case 3 =>
-      "fd" + BitcoinSUtil.encodeHex(BitcoinSUtil.toByteSeq(num).reverse)
-    case 5 => "fe" + BitcoinSUtil.encodeHex(BitcoinSUtil.toByteSeq(num).reverse)
-    case _ => "ff" + BitcoinSUtil.encodeHex(BitcoinSUtil.toByteSeq(num).reverse)
+    case 3 => "fd" + ScriptNumberUtil.longToHex(num)
```

Comment:
Don't think these should be calling `ScriptNumberUtil`, if I remember correctly the numbers are simply unsigned integers which could go through `UInt32` or `UInt64` possibly